core/zoe/zoe.chainmail

```diff
@@ -0,0 +1,150 @@
+ * they receive from someone else. 
+ */
+
+interface Zoe {
+
+  /**
+   * Get the long-lived seatAssay. The mint associated with the
+   * seatAssay creates the ERTP payments that represent the right to
+   * claim the payoffs of involvement in a contract.
+   */
+  getSeatAssay() -> (Assay);
+
+  /**
+   * Get the long-lived escrowReceiptAssay. The mint associated
+   * with the escrowReceiptAssay creates the ERTP payments that
+   * represent proof of escrow and may be passed into a
+   * governingContract method.
+   */
+  getEscrowReceiptAssay() -> (Assay);
+
+  /**
+   * Returns the array of assays for the particular instance. This is 
+   * helpful in the case that the user has forgotten the order.
+   */
+  getAssaysForInstance() -> (List(Assay));
+
+  /** 
+   * Zoe is long-lived. We can use Zoe to create governing contract
+   * instances by specifying by name a particular governing contract 
+   * library to install, as well as the array of assays for that 
+   * particular instance. We get back an instance, an id for that 
+   * instance, and the name of the library.
+   */
+  makeInstance(libraryName :Text, assays :List(Assay)) -> (InstanceInformation);
+  
+  /**
+   * Credibly get the instance from the instanceId.
+   */
+  getInstance(instanceId :Object) -> (InstanceInformation);
+
+  /** 
+   * To escrow, the user must provide an `offerDesc` which is a list of
+   * objects with a rule property (such as 'offerExactly') and an assetDesc
+   * property. The objects in the `offerDesc` must be in the same order
+   * as the assays associated with a governing contract. That is, the
+   * assetDesc in index 0 of `offerDesc` should be an assetDesc for the assay
+   * in index 0 of the assays array.`payments` is an array of the
+   * actual payments to be escrowed, following the rules in the
+   * offerDesc. If the offerDesc rule is 'offerExactly' or 'offerAtMost',
+   * then a payment matching the assetDesc is expected. If the offerDesc
+   * rule is 'wantAtLeast' or 'wantExactly' then the payment will be
+   * ignored and should be `undefined`.
+   */
+  escrow (offerDesc :List(OfferDescElement), payments :List(Payment)) -> (EscrowReceiptAndClaimPayoff);
+}
+
+/**
+ * An offer description is an array of OfferDescElements. The possible
+ * rules are 'offerExactly', 'offerAtMost', 'wantExactly', and 
+ * 'wantAtLeast'.
+ */
+struct OfferDescElement ( ) {
+  rule :Rule;
+  assetDesc :AssetDesc;
+}
+
+interface ZoeGoverningContractFacet () {
+
+  /** 
+   * Instruct Zoe to try reallocating for the given offerIds.
+   * Reallocation is a matrix (array of arrays) where the rows are the
+   * extents to be paid to the player who made the offer at the same
+   * index in the offerIds array. The reallocation will only happen if
+   * 'offer safety' and conservation of rights are true, as enforced by Zoe.
+   */
+  reallocate (instanceId :Object, offerIds :List(Object), reallocation :List(List(Extent)));
```

Comment:
I thought this was already fixed --- that the facet is specific to an instance, so the methods don't need to carry the instanceId.